contracts/src/Assets.sol

```diff
@@ -93,6 +85,45 @@ library Assets {
         IERC20(token).safeTransferFrom(sender, assetHubAgent, amount);
     }
 
+    /// @dev helper method to handle common logic for sending tokens
+    function _sendToken(
```

Comment:
I actually prefer the pre-refactored version. I think _sendToken is just different enough to warrant two different methods. This refactored method does not read as clearly. I am open to suggestions.

Reply:
I agree, I prefer two distinct cases.
The only duplicated code really is:
```
        AssetsStorage.Layout storage $ = AssetsStorage.layout();

        _transferToAgent(assetHubAgent, token, sender, amount);

        ... specific code ...

        extraFee = $.sendTokenFee;
        emit TokenSent(sender, token, destinationChain, abi.encodePacked(destinationAddress), amount);
```

I think it can be refactored to and then moved out to another method.
```
        ... specific code ...

        AssetsStorage.Layout storage $ = AssetsStorage.layout();

        _transferToAgent(assetHubAgent, token, sender, amount);

        extraFee = $.sendTokenFee;
        emit TokenSent(sender, token, destinationChain, abi.encodePacked(destinationAddress), amount);
```

The `... specific code ...` does not depend on the getting the assets storage or transferToAgent. The specific code validates the inputs and builds the payload only.

Reply:
I agree too. Lets reduce the scope of this PR to just the inbound pallet refactor.

Reply:
@alistair-singh good point, that's a great idea. I refactored it accordingly. @vgeddes I do think this a bit cleaner, let me know if you agree. https://github.com/Snowfork/snowbridge/pull/943/commits/0080d922003400814a622e1be2feaedaca82045a

Reply:
Nah, I prefer the pre-refactored version. :)

Reply:
Reverted in https://github.com/Snowfork/snowbridge/pull/943/commits/df77b6854972bb123b5f987684dc7849c8876690.